LET'S PLAY WITH **DATA

In [3]:
import pandas as pd

In [6]:
import pandas as pd
friends = {'Name':['Malay','Raghib','Pranjal','Pranav','Raza','Selena','Emma','scarlett'],
           'Marks':[80,85,90,95,87,70,75,80],
           'Gender':['Male','Male','Male','Male','Male','Female','Female','Female']
          }

#here friend is a dictionary , and using this dictionary we created a dataframe.

df = pd.DataFrame(friends)
df

Name  Marks  Gender
0     Malay     80    Male
1    Raghib     85    Male
2   Pranjal     90    Male
3    Pranav     95    Male
4      Raza     87    Male
5    Selena     70  Female
6      Emma     75  Female
7  scarlett     80  Female

In [8]:
#Q1. Display top 3 rows of the Dataset

df.head() #by-default it shows top 5 rows
df.head(3) #prints top 3 rows

Name  Marks Gender
0    Malay     80   Male
1   Raghib     85   Male
2  Pranjal     90   Male

In [9]:
#Q.2 check last 3 Rows of The DataSet

df.tail() #by-defalt it shows 5 last rows

df.tail(5) #displays last 5 rows

Name  Marks  Gender
3    Pranav     95    Male
4      Raza     87    Male
5    Selena     70  Female
6      Emma     75  Female
7  scarlett     80  Female

In [13]:
#Find shape of our Dataset (Number of Rows and Number of columns)
#shape is not a method(funciton),it is an attribute of a panda dataframe

print(df.shape) #prints tuple having no of rows and column in table

print('Number of Rows : ',df.shape[0])
print('Number of columns :',df.shape[1])

(8, 3)
Number of Rows :  8
Number of columns : 3


### Get infromation About Our Dataset like total number of rows and columns, Total number of columns , datatypes of each column and memory rquirement

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Name    8 non-null      object
 1   Marks   8 non-null      int64 
 2   Gender  8 non-null      object
dtypes: int64(1), object(2)
memory usage: 320.0+ bytes


### check Null values in the dataset

In [15]:
df.isnull()

Name  Marks  Gender
0  False  False   False
1  False  False   False
2  False  False   False
3  False  False   False
4  False  False   False
5  False  False   False
6  False  False   False
7  False  False   False

In [20]:
df.isnull().sum() #columnwise

Name      0
Marks     0
Gender    0
dtype: int64

In [19]:
df.isnull().sum(axis = 0) #displays null values column wise

Name      0
Marks     0
Gender    0
dtype: int64

In [21]:
df.isnull().sum(axis = 1) #displayes the null value row wise

0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
dtype: int64

# Get overall statistics about the dataframe

In [22]:
df.describe() #it displays on that column that contain numerical value

Marks
count   8.000000
mean   82.750000
std     8.137216
min    70.000000
25%    78.750000
50%    82.500000
75%    87.750000
max    95.000000

In [23]:
df.describe(include = 'all')

Name      Marks Gender
count       8   8.000000      8
unique      8        NaN      2
top     Malay        NaN   Male
freq        1        NaN      5
mean      NaN  82.750000    NaN
std       NaN   8.137216    NaN
min       NaN  70.000000    NaN
25%       NaN  78.750000    NaN
50%       NaN  82.500000    NaN
75%       NaN  87.750000    NaN
max       NaN  95.000000    NaN

 25% marks is below 78.75
 50% marks is below 82.5
 75% marks is below 87.75
    
    max is 95